src/xref2/tools.ml

```diff
@@ -915,6 +915,21 @@ and lookup_class_type :
   in
   res
 
+and resolve_and_lookup_parent :
+    mark_substituted:bool ->
+    add_canonical:bool ->
+    Env.t ->
+    Cpath.module_ ->
+    ( Cpath.Resolved.parent * Component.Signature.t * Component.Substitution.t,
+      [ `Parent of parent_lookup_error ] )
+    result =
+ fun ~mark_substituted ~add_canonical env parent ->
+  resolve_module ~mark_substituted ~add_canonical env parent
+  |> map_error (fun e -> `Parent (`Parent_module e))
+  >>= fun (parent, _) ->
+  lookup_parent ~mark_substituted env (`Module parent)
```

Comment:
The double-lookup seems weird at first but I agree it's a good idea to take advantage of the existing caches. This seems to retain the same performances as before with less code.

A small comment would be nice.

Reply:
Is there a double lookup? The code I removed looks identical to the code of lookup_parent (`Module case), except it uses `expansion_of_module` instead of `expansion_of_module_cached`, but `lookup_parent` is already memoized.

Reply:
`resolve_module` does a first lookup (memoized), giving a resolved path and a signature. The signature is discarded and `lookup_parent` does a second (memoized) lookup, giving the same signature and the computed substitutions.

I think this is acceptable as both lookups are memoized and we do need the resolved path from the first lookup. The unnecessary work is very small and the avoided code duplication will help us do the changes we planned to the `lookup_parent` function.